File: Lista 3/app/src/main/java/com/example/lista_3_pum/MainActivity.java
package com.example.lista_3_pum;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.NavController;
import androidx.navigation.fragment.NavHostFragment;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.bottomnavigation.BottomNavigationView;

public class MainActivity extends AppCompatActivity {

    private NavController navController;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Set up the NavHostFragment
        NavHostFragment navHostFragment = (NavHostFragment) getSupportFragmentManager().findFragmentById(R.id.fragmentContainerView);
        if (navHostFragment != null) {
            navController = navHostFragment.getNavController();
        }

        // Set up the BottomNavigationView with the NavController
        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);
        if (navController != null) {
            NavigationUI.setupWithNavController(bottomNavigationView, navController);
        }
    }
}

File: Lista 6/app/src/main/java/com/example/lista_6_pum/MainActivity.kt
package com.example.lista_6_pum

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.navigation.compose.rememberNavController
import com.example.lista_6_pum.Screens.MainScreen

class MainActivity : ComponentActivity(){
    override fun onCreate(savedInstanceState: Bundle?){
        super.onCreate(savedInstanceState)
        setContent{
            val navController = rememberNavController()
            MainScreen(navController)
        }
    }
}
